core/src/main/java/org/eclipse/hono/util/NotificationDeviceCommandReadyConstants.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 1.0 which is available at
+ * https://www.eclipse.org/legal/epl-v10.html
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ */
+
+package org.eclipse.hono.util;
+
+import java.time.Instant;
+import java.util.Optional;
+
+import org.apache.qpid.proton.message.Message;
+
+/**
+ * Constants used for command readiness notifications of a device.
+ */
+public final class NotificationDeviceCommandReadyConstants extends NotificationConstants {
+    /**
+     * Prevent construction.
+     */
+    private NotificationDeviceCommandReadyConstants() {
+    }
+
+    /**
+     * Devices that are always read to receive an upstream message can use this as value for {@link MessageHelper#APP_PROPERTY_DEVICE_TTD} to indicate it in messages.
+     */
+    public static final int TTD_VALUE_ALWAYS_READY = -1;
+
+    /**
+     * Verify if a deviceNotification event (as a JSON object that is defined in the Event API) is signalling that the
+     * device for which it was received should be ready to receive a command.
+     * This is evaluated at the point in time this method is invoked.
+     *
+     * @param tenantId Tenant this message is scoped to.
+     * @param deviceNotification Message that is evaluated.
+     * @return Boolean {@code true} if the event signals that the device now should be ready to receive a command, {@code false}
+     * otherwise.
+     * @throws NullPointerException If deviceNotification is {@code null}.
+     */
+    public static Optional<NotificationDeviceReadyToDeliver> isDeviceCurrentlyReadyForCommands(final String tenantId, final Message deviceNotification) {
```

Comment:
IMHO we do not need this in a separate class, instead this method should go into `MessageHelper` where we already have means to read and set the ttd. then remove this class ...

Reply:
Good point, will move it to the `MessageHelper`.